test/workloadapi/test_default_workload_api_client_fetch_x509_svid.py

```diff
@@ -0,0 +1,77 @@
+import pytest
```

Comment:
The name of this file is based on the previous approach to test the Workload API client, right?
With the new approach I think it should be renamed to `test_default_workload_api_client.py`

Reply:
no, I added that test file that covers the different test scenarios for that method, using mocks.

Reply:
So you plant to have a file for each Workload API method to test?

Reply:
Yes, that's the idea, separate tests in smaller and more specific modules.

Reply:
Ok, I would say that the small and specific module should be one that test the complete Workload API. 
Let's see what @scotte and @glaucimar thinks about this. I'll be ok with what the majority decides.

Reply:
Why is that? It seems OK to me to have multiple test files for testing a complex module like this one, we would have a `test_default_workload_api_client.py` for common things like configuration, and then separate test files for each workload API method. This is a practice I've seen in other languages.

Reply:
I would generally steer towards a single test file named to echo the module being tested, HOWEVER for complex modules (such as the workload API implementation), breaking out tests by function, instead of module, makes perfect sense to me, to have tests for common functionality grouped together.

Reply:
Ok, 2 votes against my poor and lonely vote 😂 
I don't see that complexity on the workload api client, it has 5 methods only but I'll trust your judgment.